### PR TITLE
fix: password reset

### LINKED ISSUE #25696: [Bug]: The new password is not recognized; Metamask only accepts the old password after wallet reset
### Describe the bug

After going through the 'Forgot password' flow in Metamask and setting a new password, Metamask throws an 'Invalid password' warning when trying to login with the new password

### Expected behavior

Metamask should only recognize the newly set password.

### Screenshots/Recordings

https://github.com/MetaMask/metamask-extension/assets/82837486/0de00905-ee01-4241-a90e-da60c5eb0976

### Steps to reproduce

1. Open the extension
2. Proceed to Forget password flow
3. Paste your secret recovery phrase
4. Set a new different password
5. Proceed to Restore your Wallet
6. After entering the account, lock it and try to login with the new password
7. It will throw an Invalid password error

### Error messages or log output

_No response_

### Version

12.0.0

### Build type

Beta

### Browser

Chrome, Firefox

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by owencraston**
## **Description**

1. What is the reason for the change?
- This fix addresses a user facing bug in production
- When a user locks the wallet and goes through the forget password flow, the wallet is not accessible with the new password. Instead the user must use the old password to unlock the wallet.
- This is a bug in the keyring controller because the encryptionKey gets generated with the old password and is not reset when the user locks the wallet and generates a new password.
2. What is the improvement/solution?
- This fix upgrades the keyring controller to version [17.1.1](https://github.com/MetaMask/core/blob/main/packages/keyring-controller/CHANGELOG.md#1711) which fixes this bug by clearing the encryption key/salt when the user locks the wallet as well as when they submit their password. This ensures that the encryption key is always in sync with the latest password. 
- The keyring controller pr for this fix can be found [here](https://github.com/MetaMask/core/pull/4514)
- This change involved a major version bump for the keyring controller from 16 to 17. [Here](https://github.com/MetaMask/core/blob/main/packages/keyring-controller/CHANGELOG.md#1711) is the changelog for this bump.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25847?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/25696

## **Manual testing steps**
- Open the extension
- Proceed to Forget password flow
- Paste your secret recovery phrase
- Set a new different password
- Proceed to Restore your Wallet
- After entering the account, lock it and try to login with the new password
- You should be able to log in with your new password

### **Before**

https://github.com/MetaMask/metamask-extension/assets/82837486/0de00905-ee01-4241-a90e-da60c5eb0976

### **After**

https://github.com/user-attachments/assets/f1f091c4-8974-415d-af28-d7b0b46184cb

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by owencraston** [comment]
@metamaskbot update-policies

**Comment by owencraston** [comment]
@metamaskbot update-policies

**Comment by owencraston** [comment]
@metamaskbot update-policies

File: test/e2e/accounts/forgot-password.spec.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { Suite } from 'mocha';
+import { unlockWallet, withFixtures, TEST_SEED_PHRASE_TWO } from '../helpers';
+import FixtureBuilder from '../fixture-builder';
+import { Driver } from '../webdriver/driver';
+
+const newPassword = 'this is the best password ever';
+
+describe('Forgot password', function (this: Suite) {
+  it('Reset password then log in with new password', async function () {
```
Comment (gantunesr) [inline test/e2e/accounts/forgot-password.spec.ts:9]: nit

```suggestion
  it('resets password and then unlock wallet with new password', async function () {
```

**Review by gantunesr — APPROVED** [review]
lgtm

**Review by mikesposito — APPROVED** [review]
Looks good!

**Comment by mikesposito** [comment]
Unfortunately, I didn't notice that this bump broke a peer dependency rule for `@metamask/signature-controller`. I opened [another PR](https://github.com/MetaMask/metamask-extension/pull/25941) to bump `@metamask/signature-controller` and `@metamask/user-operation-controller`

**Pull request merged**